internal/pkg/describe/lb_web_service.go

```diff
@@ -289,36 +300,79 @@ type WebServiceRoute struct {
 	URL         string `json:"url"`
 }
 
-// ServiceDiscovery contains serialized service discovery info for an service.
+// ServiceDiscovery contains serialized service discovery info for an ECS service.
 type ServiceDiscovery struct {
 	Environment []string `json:"environment"`
-	Namespace   string   `json:"namespace"`
+	// Namespace is not an accurate name but unfortunately we can't take it back.
+	Namespace string `json:"namespace"`
 }
 
 type serviceDiscoveries []*ServiceDiscovery
 
-func (s serviceDiscoveries) humanString(w io.Writer) {
-	headers := []string{"Environment", "Namespace"}
+// ServiceConnect contains serialized service connect info for an ECS service.
+type ServiceConnect struct {
+	Environment []string `json:"environment"`
+	DNSName     string   `json:"dnsName"`
+}
+
+type serviceConnects []*ServiceConnect
+
+type serviceEndpoints struct {
+	discoveries serviceDiscoveries
+	connects    serviceConnects
+}
+
+func (s serviceEndpoints) humanString(w io.Writer) {
+	headers := []string{"Environment", "Endpoint", "Type"}
 	fmt.Fprintf(w, "  %s\n", strings.Join(headers, "\t"))
 	fmt.Fprintf(w, "  %s\n", strings.Join(underline(headers), "\t"))
-	for _, sd := range s {
-		fmt.Fprintf(w, "  %s\t%s\n", strings.Join(sd.Environment, ", "), sd.Namespace)
+	scEnvToEndpoints := make(map[string]map[string]struct{})
```

Comment:
I feel like our data structures are pretty complicated 💭  something is wrong with how we're modeling data, and we have to duplicate the same marshaling logic for both SC and SD.

Let me know what you think of the above comment for an alternative API.

Reply:
this is just a map of string and set...i think it is still required for doing aggregation even if we adopt the alternative api.